include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1017 @@
+            {
+                using dbl = std::numeric_limits< float_64 >;
+                outFile.precision( dbl::digits10 );
+                if (currentStep > 0.0){
+                    int startWindow_y = movingWindow.globalDimensions.offset.y();
+                    int endWindow_y = movingWindow.globalDimensions.size.y() + startWindow_y;
+                    if (fisttimestep == true) 
+                    {
+                        outFile << "#step emit_all " << std::scientific;
+                        for (int i = startWindow_y; i < (endWindow_y + 10); i+=10)
+                        {
+                            outFile << i * UNIT_LENGTH * SI::CELL_WIDTH_SI;
+                            if( i < (endWindow_y) )
+                            {
+                                outFile << " ";
+                            }
+                        }
+                        outFile << std::endl;
+                        fisttimestep = false;
+                    } 
+                    outFile << currentStep << " "
+                            << std::scientific;
+
+                    long double numElec_all = 0.0;
+                    long double ux2_all = 0.0;
+                    long double pos2_SI_all = 0.0;
+                    long double xux_all = 0.0;
+
+                    for (int i = startWindow_y; i < endWindow_y; i++)
+                    {
+                            numElec_all+= static_cast<long double>(globalCount_e.getDataPointer( )[i]);
+                            ux2_all    += static_cast<long double>(globalSumMom2.getDataPointer( )[i]) * UNIT_MASS* UNIT_MASS /(SI::ELECTRON_MASS_SI*SI::ELECTRON_MASS_SI);
+                            pos2_SI_all+= static_cast<long double>(globalSumPos2.getDataPointer( )[i]) *UNIT_LENGTH*UNIT_LENGTH ;
+                            xux_all    += static_cast<long double>(globalSumMomPos2.getDataPointer( )[i])*UNIT_MASS *UNIT_LENGTH /SI::ELECTRON_MASS_SI;
+                    }
+                    double emit_all = algorithms::math::sqrt(((double)pos2_SI_all * (double)ux2_all - (double)xux_all * (double)xux_all)/((double)numElec_all*(double)numElec_all));
+                    if (emit_all > 0.0){
+                        outFile << emit_all << " ";
+                    }
+                    else {
+                        outFile << "0.0 ";
+                    }
+
+                    for (int i = startWindow_y; i < endWindow_y; i+=10)
+                    {
+                        double numElec   = globalCount_e.getDataPointer( )[i];
+                        double mom2_SI   = globalSumMom2.getDataPointer( )[i] * UNIT_MASS * UNIT_SPEED* UNIT_MASS * UNIT_SPEED;
+                        double pos2_SI   = globalSumPos2.getDataPointer( )[i] *UNIT_LENGTH*UNIT_LENGTH ;
+                        double mompos_SI = globalSumMomPos2.getDataPointer( )[i]*UNIT_MASS * UNIT_SPEED*UNIT_LENGTH;
+                        for ( int j=i+1; j<i+9 && j < endWindow_y; j++ ){
```

Comment:
This line will always leave out the value `i+9` it must be `j<i+10`

Reply:
I think we should skip that functionality altogether, can be done cheaply in post-processing

Reply:
I agree with @ax3l  the resulting data set is very small and post processing smoothing will be not hard and give more degree of freedom to select the operation to smooth the data.